mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -252,7 +254,13 @@ class LogisticRegression(override val uid: String)
 
     if (handlePersistence) instances.unpersist()
 
-    copyValues(new LogisticRegressionModel(uid, weights, intercept))
+    val model = copyValues(new LogisticRegressionModel(uid, weights, intercept))
```

Comment:
Can you please explain why this `copyValues` is necessary? and I'm unable to understand how $(probabilityCol) gives a string because when I do this.

    val model = lr.fit(dataset)
    $(lr.probabilityCol)

I get 

    error: not found: value $
    $(probabilityCol)